detectron2/modeling/proposal_generator/proposal_utils.py

```diff
@@ -167,14 +166,15 @@ def add_ground_truth_to_proposals_single_image(gt_boxes, proposals):
     Returns:
         Same as `add_ground_truth_to_proposals`, but for only one image.
     """
+    gt_boxes = gt.gt_boxes
     device = proposals.objectness_logits.device
     # Assign all ground-truth boxes an objectness logit corresponding to
     # P(object) = sigmoid(logit) =~ 1.
     gt_logit_value = math.log((1.0 - 1e-10) / (1 - (1.0 - 1e-10)))
     gt_logits = gt_logit_value * torch.ones(len(gt_boxes), device=device)
 
     # Concatenating gt_boxes with proposals requires them to have the same fields
-    gt_proposal = Instances(proposals.image_size)
+    gt_proposal = Instances(proposals.image_size, **gt.get_fields())
```

Comment:
This still fails if `gt` has fields not predicted by the RPN (e.g. `gt_masks`, `gt_boxes`), or your custom RPN predicts fields that are not in gt.

The correct way would be to: for every field in `proposals` are not `proposal_boxes/objectness_logits`, use the field with the same name in `gt`. If the field does not exist in `gt`, throws an error.